homeassistant/components/swidget/light.py

```diff
@@ -0,0 +1,96 @@
+"""Support for Swidget lights."""
+
+from __future__ import annotations
+
+import logging
+import math
+from typing import Any, cast
+
+from swidget.swidgetdimmer import SwidgetDimmer
+import voluptuous as vol
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.color import brightness_to_value, value_to_brightness
+
+from .const import DOMAIN
+from .coordinator import SwidgetDataUpdateCoordinator
+from .entity import CoordinatedSwidgetEntity, async_refresh_after
+
+_LOGGER = logging.getLogger(__name__)
+
+BRIGHTNESS = "brightness"
+BRIGHTNESS_SCALE = (0, 100)
+VAL = vol.Range(min=0, max=100)
+SWIDGET_SET_BRIGHTNESS_SCHEMA = cv.make_entity_service_schema({BRIGHTNESS: VAL})
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up light."""
+    coordinator: SwidgetDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    if coordinator.device.is_dimmer:
```

Comment:
what if the device is not a dimmer? It doesn't have any entities? Why even add it at all

Reply:
Swidget (currently) have three types of devices: dimmer, outlet, on/off switch. We only set up the light entity if the device is a dimmer